client/lib/mention.js

```diff
@@ -0,0 +1,71 @@
+var hueHash = require('./hue-hash')
+
+/**
+ * Determines if name contains the characters in the partial name, in order.
+ */
+module.exports.containsSubseq = function(name, part) {
+  var offset = 0
+  var remain = part
+  var nexdex
+  while (remain !== "") {
+    nexdex = name.indexOf(remain.substr(0, 1), offset)
+    if (nexdex < 0) {
+      return false
+    }
+    offset = nexdex + 1
+    remain = remain.substr(1)
+  }
+  return true
+}
+
+/**
+ * From a name and a partial produce a score. A score of
+ * zero indicates "no match whatsoever", all other scores
+ * are positive.
+ */
+module.exports.scoreMatch = function(name, part) {
+  // FIXME Use proper Unicode-aware case-folding, if not already
+  var partCf = part.toLowerCase()
```

Comment:
I originally had `name_cf` and the linter was annoyed. (In the original it was `name-cf`.) Short names would be nice here because they allow you to see the parallels between the case-folded and original checks, but I can live with `nameLowercase` if it would otherwise be hard to read.